tests/homework2/task2/test_cache.py

```diff
@@ -0,0 +1,218 @@
+from collections import OrderedDict
+from unittest import TestCase
+from homework.homework2.task2.cache import cache_decorator
+
+
+class CacheFunctionWithOneArgumentTestCase(TestCase):
+    def test_negative_size_cache(self):
+        with self.assertRaises(ValueError) as context:
+
+            @cache_decorator(size=-162)
+            def fib(n: int) -> int:
+                if n == 1 or n == 2:
+                    return 1
+                return fib(n - 1) + fib(n - 2)
+
+        self.assertTrue("Size must be a non-negative number." in str(context.exception))
+
+    def test_zero_size_cache(self):
+        @cache_decorator
+        def fib(n: int) -> int:
+            if n == 1 or n == 2:
+                return 1
+            return fib(n - 1) + fib(n - 2)
+
+        fib(5)
+        fib(6)
+        fib(7)
+
+        self.assertFalse(fib._cache)
+
+    def test_positive_size_cache(self):
+        @cache_decorator(size=10)
+        def fib(n: int) -> int:
+            if n == 1 or n == 2:
+                return 1
+            return fib(n - 1) + fib(n - 2)
+
+        fib(10)
+        fib(9)
+        fib(8)
+
+        print(fib._cache)
+
+        self.assertEqual(
+            fib._cache,
+            OrderedDict(
+                [
+                    ((2,), 1),
+                    ((1,), 1),
+                    ((3,), 2),
+                    ((4,), 3),
+                    ((5,), 5),
+                    ((6,), 8),
+                    ((7,), 13),
+                    ((8,), 21),
+                    ((9,), 34),
+                    ((10,), 55),
+                ]
+            ),
+        )
+
+
+class CacheFunctionWithArgsTestCase(TestCase):
+    def test_negative_size_cache(self):
+        with self.assertRaises(ValueError) as context:
+
+            @cache_decorator(size=-444)
+            def sum_of_squares(*args: int) -> int:
+                return sum(x ** 2 for x in args)
+
+        self.assertTrue("Size must be a non-negative number." in str(context.exception))
```

Comment:
Also, you use this error message in several places in the tests. Maybe it would be better to add a const variable with this value?